checkoutmanager/config.py

```diff
@@ -116,6 +116,12 @@ def directories(self, group=None):
                 result.append(dirinfoclass(directory, url))
         return sorted(result)
 
+    def directory_from_url(self, url):
+        return [x for x in self.directories() if x.url == url][0]
```

Comment:
If nothing is found, this gives you an unclear generic IndexError. Handier to just return nothing and let the caller deal with it.

    for directory in self.directories():
        if directory.url == url:
            return directory

It is a little bit more verbose, but also more readable in this case.